src/proposer/proposer.cpp

```diff
@@ -178,13 +219,26 @@ class ProposerImpl : public Proposer {
     m_waiter.Wake();
   }
 
+  std::shared_ptr<const CBlock> GenerateBlock(CWallet &wallet,
```

Comment:
Does it make sense to explicitly call GenerateBlock on ActiveProposer? 
For me it seems like an error - ActiveProposer should figure it out itself when to propose. And in functional tests I would prefer to have an error when I try to explicitly generate blocks on node with `-proposing=1`

Reply:
Are you saying that when `mine_blocks_on_demand = true` then auto-proposing should not work?

Reply:
No, when `-proposing=1` and `mine_blocks_on_demand` - manual propose should not work.
But your variant also sounds reasonable

Reply:
When `-proposing=1` then `propose` and `proposetoaddress` should not work?

That makes somewhat sense to me.

Reply:
> When -proposing=1 then propose and proposetoaddress should not work?
Yes, right

Reply:
I've added something like:
```
if (m_settings->node_is_proposer) {
      throw JSONRPCError(RPC_INTERNAL_ERROR, "Node is automatically proposing.");
    }
```

Reply:
If we agree that when `-proposing=1` `propose` should be an error, doesn't this make the whole `ActiveProposerImpl::GenerateBlock` meaningless?

Reply:
Well, they keep the same interface, also you don't wanna replicate code.

Reply:
We can just throw not-implemented-exception there instead of implementing it

Reply:
@AM5800 @Gnappuraz it sounds reasonable to not implement `GenerateBlock` as it should never be called. And then we can have a nicer interface for `GenerateBlock`, it would accept always `CScript` instead of `boost::optional<CScript>`